### PR TITLE
Use buffered output for index writer

### PR DESCRIPTION
**Pull request by navis**
Use buffered stream whenever possible.

### CODE REVIEW

File: indexing-hadoop/src/main/java/io/druid/indexer/Utils.java [hunk 1]
```diff
@@ -45,6 +46,17 @@
   public static OutputStream makePathAndOutputStream(JobContext job, Path outputPath, boolean deleteExisting)
       throws IOException
   {
+    return makePathAndOutputStream(job, outputPath, deleteExisting, -1);
```
Comment (nishantmonu51) [inline indexing-hadoop/src/main/java/io/druid/indexer/Utils.java]: can we have two separate methods makePathAndOutputStream and makePathAndBufferedOutputStream that takes buffer size  ?

**Comment by nishantmonu51** [comment]
can you also post any numbers on how much performance improvements you are seeing after this change ?

**Comment by drcrallen** [comment]
Does this add pressure on the heap during indexing time?

**Comment by navis** [comment]
@nishantmonu51 Sadly, even with change of streaming copy for files to channel transfer, I couldn't see any notable performance improvement. Should we close this?
@drcrallen If we have a lot of dimensions, it could be a problem. Anyway, TmpFileIOPeon makes 8K buffered stream for each file currently. Now we can adjust the size appropriately for cases.

**Comment by binlijin** [comment]
I also have test TmpFileIOPeon with different buffer size to improve index merger performance,  but i find with small data the performance can increase, but with big data the performance just the same and find the bottleneck is CPU.

**Comment by navis** [comment]
@nishantmonu51, @drcrallen, @binlijin Ok, then let's close this. But channel transfer one would be good for CPU usage perspect even the performance is not improved. I'll PR it after closing this. Thanks.

**Pull request closed without merging**